src/harlequin/autocomplete/completers.py

```diff
@@ -74,6 +85,21 @@ def extend_catalog(self, parent: CatalogItem, items: list[CatalogItem]) -> None:
             self._extra_completions,
         )
 
+    @staticmethod
+    def _fuzzy_match(
+        match_val: str, completions: list[HarlequinCompletion]
+    ) -> list[HarlequinCompletion]:
+        regex_base = ".{0,2}?".join(f"({re.escape(c)})" for c in match_val)
+        regex = "^.*" + regex_base + ".*$"
+        match_regex = re.compile(regex, re.IGNORECASE)
+        matches = [c for c in completions if match_regex.match(c.match_val)]
+
+        # Sort in ascending length.
+        # I am assuming here that more insertions are less likely to be
+        # the "right" match.
+        matches.sort(key=lambda c: len(c.match_val), reverse=True)
```

Comment:
```suggestion
        matches.sort(key=lambda c: len(c.match_val))
```
From the snapshot tests, I think adding the `reverse` causes the longest fuzzy matches to be listed first, which is the opposite of what you want.
![image](https://github.com/user-attachments/assets/5bcd86a7-1a89-44f7-900a-83b863805035)

Reply:
Good catch